packages/blade/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable @typescript-eslint/prefer-ts-expect-error */
+import React from 'react';
+import { ThumbIcon } from './ThumbIcon';
+import type { SwitchProps } from './types';
+import { switchHoverVariants } from './switchTokens';
+import { Thumb } from './Thumb';
+import { AnimatedThumb } from './AnimatedThumb';
+import { SwitchTrack } from './SwitchTrack';
+import { useTheme } from '~components/BladeProvider';
+import BaseBox from '~components/Box/BaseBox';
+import { useCheckbox } from '~components/Checkbox/useCheckbox';
+import { SelectorInput } from '~components/Form/Selector/SelectorInput';
+import {
+  assignWithoutSideEffects,
+  makeAccessible,
+  metaAttribute,
+  useBreakpoint,
+  usePrevious,
+} from '~utils';
+import type { BladeElementRef } from '~src/hooks/types';
+import { SelectorLabel } from '~components/Form/Selector/SelectorLabel';
+
+const _Switch: React.ForwardRefRenderFunction<BladeElementRef, SwitchProps> = (
+  {
+    defaultChecked,
+    isChecked,
+    isDisabled,
+    name,
+    onChange,
+    size = 'medium',
+    value,
+    accessibilityLabel,
+    id,
+    testID,
+  },
+  ref,
+): React.ReactElement => {
+  const { state, inputProps } = useCheckbox({
+    role: 'switch',
+    defaultChecked,
+    isChecked,
+    isIndeterminate: false,
+    hasError: false,
+    hasHelperText: false,
+    isDisabled,
+    isRequired: false,
+    name,
+    value,
+    onChange,
+  });
+
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const previousChecked = usePrevious(state.isChecked);
+
+  return (
+    <BaseBox
```

Comment:
Should we make `cursor: pointer` for Switch?

Reply:
Might need some design inputs here, but cursor: pointer is kind of covering a bit too much of the switch's visuals